specs/requests/artifacts_spec.py

```diff
@@ -133,6 +133,73 @@
 
+        with description('UPDATE MANY'):
+            with context('valid request'):
+                with context("all the resources exist"):
+                    with before.each:
+                        with Mock() as elastic_mock:
+                            for i in range(2):
+                                elastic_mock.get(
+                                    doc_type='_all', id=f'{get_uuid(i)}',
+                                    index='artifact').returns(es_get_response(i))
+                            elastic_mock.update(
+                                doc_type='image', id=f'{get_uuid(0)}', index='artifact',
+                                body={'doc': {
+                                    "updated_at": matches_regexp(date_regex()),
+                                    "tags": ["blue", "red"]}})
+                            elastic_mock.update(
+                                doc_type='image', id=f'{get_uuid(1)}', index='artifact',
+                                body={'doc': {
+                                    "updated_at": matches_regexp(date_regex()),
+                                    "tags": ["blue", "green"]}})
+
+                        current_app.es = elastic_mock
+                        self.response = self.context.client().patch(f"/images", json={
+                            "data": [
+                                {
+                                    "id": get_uuid(0),
+                                    "tags": ["blue", "red"]
+                                },
+                                {
+                                    "id": get_uuid(1),
+                                    "tags": ["blue", "green"]
+                                }
+                            ]
+                        })
+
+                    with it('returns a 204 status code'):
+                        expect(self.response.status_code).to(equal(204))
+
+                with context("a resource does not exists"):
+                    with before.each:
+                        with Mock() as elastic_mock:
+                            elastic_mock.get(
+                                doc_type='_all', id=f'{get_uuid(0)}',
+                                index='artifact').raises(NotFoundError)
+                        current_app.es = elastic_mock
+                        self.response = self.context.client().patch(f"/images", json={
+                            "data": [{ "id": get_uuid(0), "tags": ["blue", "red"]}]
+                        })
+
+                    with it('returns a 404 status code'):
+                        expect(self.response.status_code).to(equal(404))
+
+                    with it('includes error message'):
+                        expect(self.response.json).to(have_key("error"))
+
+            with context('invalid request'):
+                with before.each:
+                    self.response = self.context.client().patch("/images", json={
+                        "data": [{}]
```

Comment:
what happens if you have a uuid as a key here?

Reply:
I think the world would be a better place...